feat: support draft APIs ZMQ_HELLO_MSG, ZMQ_DISCONNECT_MSG and ZMQ_HICCUP_MSG

simeyc opened the pull request:
Features:
- Add Set methods for additional DRAFT APIs:
  - `SetHelloMsg`
  - `SetDisconnectMsg`
  - `SetHiccupMsg`

pebbe commented:
This needs some work.

`ZMQ_HELLO_MSG` and `ZMQ_DISCONNECT_MSG` were introduced in 0MQ version 4.3.3

`ZMQ_HICCUP_MSG` was introduced in 0MQ version 4.3.5

So, in all changed files, you need to add checks for the patch version.

In `draft/zmg4.go` you need to add definitions for `ErrorNotImplemented433` and `ErrorNotImplemented435`

simeyc commented:
> This needs some work.
> 
> `ZMQ_HELLO_MSG` and `ZMQ_DISCONNECT_MSG` were introduced in 0MQ version 4.3.3
> 
> `ZMQ_HICCUP_MSG` was introduced in 0MQ version 4.3.5
> 
> So, in all changed files, you need to add checks for the patch version.
> 
> In `draft/zmg4.go` you need to add definitions for `ErrorNotImplemented433` and `ErrorNotImplemented435`

Thanks, I have implemented the changes.

It seems that libzmq's `getsockopt` doesn't support the new options; it returns `EINVAL` for options `ZMQ_HELLO_MSG`, `ZMQ_DISCONNECT_MSG`, and `ZMQ_HICCUP_MSG`, and they are absent from [the docs](https://libzmq.readthedocs.io/en/latest/zmq_getsockopt.html), so I removed the getters from `draft/socketget.go`.
